File: Data Structures & Algorithms/palindromic-substrings/submission-0.java
class Solution {
    public int countSubstrings(String s) {
        int ans = 0;
        int n = s.length();

        // find even
        for(int i=0;i<n-1;i++){
            int left = i;
            int right = i+1;
            while(s.charAt(left)==s.charAt(right)){
                ans++;
                if(left<=0 || right>=n-1)break;
                left--;
                right++;
            }
        }
        // find odd
        for(int i=0;i<n;i++){
            int left = i;
            int right = i;
            while(s.charAt(left)==s.charAt(right)){
                ans++;
                if(left<=0 || right>=n-1)break;
                left--;
                right++;
            }
        }
        return ans;
    }
}

File: Data Structures & Algorithms/find-k-closest-elements/submission-0.cpp
class Solution {
public:

    vector<int> findClosestElements(vector<int>& arr, int k, int x) {
        vector<int> ans;
        auto ind = lower_bound(arr.begin(),arr.end(),x);
        auto left = ind-1;
        auto right = ind;
        while(k>0 && left>=arr.begin() && right<arr.end()){
            if(abs(x-*left)<=abs(x-*right)){
                ans.push_back(*left);
                left--;
            }
            else{
                ans.push_back(*right);
                right++;
            }
            k--;
        }
        while(k>0 && left>=arr.begin()){
            ans.push_back(*left);
            left--;
            k--;
        }
        while(k>0 && right<arr.end()){
            ans.push_back(*right);
            right++;
            k--;
        }
        sort(ans.begin(),ans.end());
        return ans;
    }
};

